scionlab/tests/test_certificates.py

```diff
@@ -81,93 +72,66 @@ def ia(as_id):
                 else:
                     AS.objects.get(as_id=as_id).delete()
 
-            trc = utils.check_trc_and_certs(self,
-                                            isd_id,
-                                            {ia(as_id) for as_id in expected_set},
-                                            expected_version=expected_version,
-                                            prev_trc=prev_trc)
-            expected_version += 1
-            prev_trc = trc
+            # Skip check and version increment if expected_set is empty -- no TRCs are created
```

Comment:
It can be empty if you enter the loop body with one element in `expected_set`. So it's not empty, so you randomly choose the "add" or the "remove" branch. If we picked remove, it will be empty now.